### DISCUSSION TITLE
Password Regex Not Enforcing Rules on Password Reset

### DISCUSSION THREAD

**Discussion by hmseeb** [pocketbase/pocketbase#6048] [Q&A]
I’ve set the following regex `^[A-Za-z\d@$!%*?&#]{8,}$` for my password field. However, when I reset my password or change it from the dashboard, it bypasses the regex validation. Can anyone point out where I might be going wrong?

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it.

Tested it locally with PocketBase v0.23.4 both from the Dashboard (aka. user update) and via the [Confirm password reset Web API](https://pocketbase.io/docs/api-records/#password-reset) and both times the password regex validator is trigered. We Also have automated tests that should verify the correct behavior:
https://github.com/pocketbase/pocketbase/blob/c6695b6a75f5daa2e156facb3cd276875676b82d/core/field_password_test.go#L253-L272

Make sure that your regex does what you intend to and that you are defining it for the correct auth collection.

If this doesn't help, then please elaborate how you are testing (aka. with what plain text password) and what are the password rules you want to enforce.

**Reply by ganigeorgiev** [reply]
_Side-note: you can test your regex for example with https://regex101.com/ but make sure to switch to the Go flavor from the sidebar._

**Reply by hmseeb** [reply]
Minimum eight characters, at least one uppercase letter, one lowercase letter, one number and one special character. 

```^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]).{8,}$```

This seems to work in general but when testing with <https://regex101.com> using golang flavor. I'm getting the following errors: 

All the errors detected are listed below, from left to right, as they appear in the pattern.
? The preceding token is not quantifiable
? The preceding token is not quantifiable
? The preceding token is not quantifiable
? The preceding token is not quantifiable

Can you help me sort this out? I don't know what's wrong with my syntax.

**Reply by ganigeorgiev** [reply]
There is no easy way to express what you want with the std Go regex (_or at least I'm not sure how_) because there is no support for lookahead/lookbehind. A workaround could be to manually list all possible group pairs but this can quickly get out of hands (_in your case with the 4 groups it will result in 24 unique pairs?_).

My advice would be to enforce the password regex rules on client-side level (if needed at all) OR if you really want to do it server-side then you can register a hook and perform the check manually without using regex but the latter is more involving and the cost of its complexity is not worth the effort in my opinion.
